src/main/kotlin/no/nav/sifinnsynapi/drift/DriftController.kt

```diff
@@ -0,0 +1,87 @@
+import no.nav.sifinnsynapi.soknad.DebugDTO
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus.OK
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+
+@RestController
+@RequiredIssuers(
+    ProtectedWithClaims(issuer = Issuers.AZURE)
+)
+class DriftController(
+    private val driftService: DriftService,
+    private val auditlogger: Auditlogger,
+    private val tokenValidationContextHolder: TokenValidationContextHolder,
+    private val oppslagsService: OppslagsService
+) {
+    companion object {
+        val logger = LoggerFactory.getLogger(DriftController::class.java)
+    }
+
+    @PostMapping("/debug$SØKNAD", produces = [MediaType.APPLICATION_JSON_VALUE], consumes = [MediaType.APPLICATION_JSON_VALUE])
+    @ResponseStatus(OK)
+    fun debugSøknader(@RequestBody debugForespørsel: DebugSøknadForespørsel): List<DebugDTO> {
+        val identTilInnloggetBruker: String = tokenValidationContextHolder.tokenValidationContext.firstValidToken.get().jwtTokenClaims.getStringClaim("NAVident")
+
+        val søkerAktørId = oppslagsService.hentIdenter(
+            HentIdenterForespørsel(
+                identer = listOf(debugForespørsel.søkerNorskIdentitetsnummer),
+                identGrupper = listOf(IdentGruppe.AKTORID)
+            )).first().identer.first().ident
+
+        val pleietrengendeAktørIder = oppslagsService.hentIdenter(
+            HentIdenterForespørsel(
+                identer = debugForespørsel.pleietrengendeNorskIdentitetsnummer,
+                identGrupper = listOf(IdentGruppe.AKTORID)
+            )).map { it.identer.first().ident }
+
+        auditLogg(uri = "/debug$SØKNAD", innloggetIdent = identTilInnloggetBruker, berørtBrukerIdent = søkerAktørId)
+        return driftService.slåSammenSøknadsopplysningerPerBarn(søkerAktørId, pleietrengendeAktørIder)
+    }
+
+    private fun auditLogg(uri: String, innloggetIdent: String, berørtBrukerIdent: String) {
+        auditlogger.logg(
+            Auditdata(
+                AuditdataHeader.Builder()
+                    .medVendor(auditlogger.vendor)
+                    .medProduct(auditlogger.product)
+                    .medSeverity("INFO")
+                    .medName("Debug sammenslåtte søknadsopplysninger")
+                    .medEventClassId(EventClassId.AUDIT_ACCESS)
+                    .build(),
+                setOf(
+                    CefField(CefFieldName.EVENT_TIME, LocalDateTime.now().toEpochSecond(ZoneOffset.UTC) * 1000L),
+                    CefField(CefFieldName.REQUEST, uri),
+                    CefField(CefFieldName.USER_ID, innloggetIdent),
+                    CefField(CefFieldName.BERORT_BRUKER_ID, berørtBrukerIdent)
```

Comment:
Det gis minimalt innsyn til pleietrengendes data (omsorgstilbud). Dette er en personopplysning som ideelt sett skal logges på pleietrengende også. Kanskje med tittel "Debug sammenslåtte søknadsopplysninger, omsorgstilbud".

Reply:
Vi bør bruke "AUDIT_SEARCH" for pleietrengendeloggingen siden innsynet er minimalt.

Reply:
Det blir en løkke for alle pleietrengende.

Reply:
👍